sdc/tests/test_series.py

```diff
@@ -2783,29 +2781,24 @@ def test_impl(S1, S2):
                         np.float32(3), np.float32(4), np.float32(5)])
         pd.testing.assert_series_equal(hpat_func(S1, S2), test_impl(S1, S2))
 
-    @skip_numba_jit
     def test_series_combine_assert1(self):
         def test_impl(S1, S2):
             return S1.combine(S2, lambda a, b: 2 * a + b)
         hpat_func = self.jit(test_impl)
 
         S1 = pd.Series([1, 2, 3])
         S2 = pd.Series([6., 21., 3., 5.])
-        with self.assertRaises(AssertionError):
-            hpat_func(S1, S2)
+        pd.testing.assert_series_equal(hpat_func(S1, S2), test_impl(S1, S2))
```

Comment:
General comment for tests: not all combinations of input series dtypes and fill_value are tested e.g. the one I mentioned before - where float fill_value is assigned to otherwise int series. There are no tests with series with non-default indexes (we refer to samelen, but it's not fully correct - series may have same len, but not same indexes), and no tests for checking func impact on result dtype, so it's hard to see from such tests what's really tested and what is not. So the suggestion is to organize tests in a different manner:

1. product of diff series dtypes (default int, int64, float64),
   same series indexes (but not same series sizes),
   fill_value is specified and of different dtypes (None, np.nan, 4, 4.2)
   Covers: test_series_combine_value_samelen
2. 
   product of diff series dtypes (default int, int64, float64),
   same series indexes (but not same series sizes),
   with fill_value is omitted
   Covers: test_series_combine_float3264, test_series_combine_integer_samelen, test_series_combine_samelen, test_series_combine_different_types
3. 
   product of diff series dtypes (default int, int64, float64),
   series indexes that align with and without -1 in indexers
   fill_value is specified and of different dtypes (None, np.nan, 4, 4.2)
   Covers: test_series_combine_integer, test_series_combine_value
4. 
   product of diff series dtypes (default int, int64, float64),
   series indexes that align with and without -1 in indexers 
   fill_value is omitted
   Covers: test_series_combine, test_series_combine_assert1, test_series_combine_assert2, test_series_combine_different_types

New test:
5. (for testing func changes dtype properly)
    product of diff series dtypes (default int, int64, float64),
    same series indexes (but not same series sizes),
    fill_value = 0
    with diff functions (chaning and not chaning res dtype e.g. preserving int domain, e.g. ** and + and not, e.g. /)

Reply:
For example, test 1 can look like this (it can also be split into two: one when we use check_dtype=False and one when we don't):
```
    def test_series_combine_same_index_fill_value(self):
        def test_impl(S1, S2):
            return S1.combine(S2, lambda a, b: 2 * a + b)
        hpat_func = self.jit(test_impl)

        n = 11
        np.random.seed(0)
        A = np.random.randint(-100, 100, n)
        B = np.arange(n) * 2 + 1
        series_index = 1 + np.arange(n)

        series_dtypes = [None, np.int64, np.float64]
        fill_values = [None, np.nan, 4, 4.2]
        for dtype1, dtype2, fill_value in product(series_dtypes, series_dtypes, fill_values):
            S1 = pd.Series(A, index=series_index, dtype=dtype1)
            S2 = pd.Series(B, index=series_index, dtype=dtype2)
            with self.subTest(S1_dtype=dtype1, S2_dtype=dtype2, fill_value=fill_value):
                result = hpat_func(S1, S2)
                result_ref = test_impl(S1, S2)
                # check_dtype=False due to difference to pandas in some cases
                pd.testing.assert_series_equal(result, result_ref, check_dtype=False)
```